configs/loader.py

```diff
@@ -0,0 +1,73 @@
+import json
+
+
+def contains(keys, dictionary):
+    for key in keys:
+        if key not in dictionary:
+            return False
+    return True
+
+
+def from_request(request):
+    data = request.get_json() or request.data.decode('utf-8')
+    if data is None or len(data) == 0:
+        error_msg = json.dumps({'success': False,
+                                'error': 'Empty config',
+                                'info': request})
+        raise Exception(error_msg)
+
+    try:
+        config = json.loads(data)
+    except json.JSONDecodeError:
+        error_msg = json.dumps({'success': False,
+                                'error': 'Json decode error',
+                                'info': data})
+        raise Exception(error_msg)
+
+    if not contains(['workspaces', 'prices'], config):
+        error_msg = json.dumps({'success': False,
+                                'error': 'Missing config info',
+                                'info': config})
+        raise Exception(error_msg)
+
+    req_workspace_keys = ['url', 'id', 'type', 'name', 'token']
+    for workspace in config['workspaces']:
+        if not contains(req_workspace_keys, workspace):
+            error_msg = json.dumps({'success': False,
```

Comment:
Shouldn't we report back which workspace fields were missing?

Reply:
Agreed, implemented.